### ISSUE TITLE
Finetune on MPS support

### ISSUE THREAD

**Issue by cocktailpeanut** [SWivid/F5-TTS#150]
I was looking into the "Auto Settings" button, which triggers the code here: https://github.com/SWivid/F5-TTS/blob/main/finetune_gradio.py#L471-L472

I wanted to add MPS support, something like:

```
if torch.cuda.is_available():
    gpu_properties = torch.cuda.get_device_properties(0)
    total_memory = gpu_properties.total_memory / (1024 ** 3)  
elif torch.backends.mps.is_available():
    total_memory = psutil.virtual_memory().available / (1024 ** 3)
```

But then I looked at the code after that, and realized I don't really understand what's going on here https://github.com/SWivid/F5-TTS/blob/main/finetune_gradio.py#L474-L481 Is this assuming there are 8 GPUs? @lpscr

Anyway, bottom line, would be nice to have MPS support if it's not too difficult. Thank you.

**Comment by SWivid** [comment]
will close this issue, feel free to open if further questions~
